pkg/system/phase4_configuring.go

```diff
@@ -1338,19 +1338,27 @@ func (r *Reconciler) ReconcileBucketLoggingPVC() error {
 	}
 
 	util.KubeCheck(r.BucketLoggingPVC)
-	if r.BucketLoggingPVC.UID != "" {
-		log.Infof("BucketLoggingPVC %s already exists. skipping ReconcileBucketloggingPVC", r.BucketLoggingPVC.Name)
-		return nil
-	}
-
 	if err := r.prepareBucketLoggingPVC(); err != nil {
 		return err
 	}
-
 	r.Own(r.BucketLoggingPVC)
-	// TODO: return only for the specific error, like RWX mode not supported by provisioner
-	if err := r.Client.Create(r.Ctx, r.BucketLoggingPVC); err != nil {
-		return err
+
+	if r.BucketLoggingPVC.UID != "" && r.BucketLoggingPVC.Status.Phase == corev1.ClaimPending {
+		log.Infof("BucketLoggingPVC %s is in pending state. Investigating the reason and updating pvc", r.BucketLoggingPVC.Name)
+		// TODO: fetch reason and return only for the specific error, like RWX mode not supported by provisioner
+
+		log.Infof("Deleting and recreating the pending BucketLoggingPVC %s", r.BucketLoggingPVC.Name)
```

Comment:
Why do we have two log info back to back? One is asking for investigating the reason and other is saying deleting and recreating?

Reply:
Actually, as you can see the comment we need to fetch events on pvc here and we should return an error if error related to `RWX` mode not supported.